src/main/java/core/basesyntax/serviceimpl/StorageFillingImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.HandlerReceiver;
+import core.basesyntax.service.StorageFilling;
+
+public class StorageFillingImpl implements StorageFilling {
+    private static final int OPERATION = 0;
+    private static final int FRUIT = 1;
+    private static final int QUANTITY = 2;
+    private static final String COMA = ",";
+    private final HandlerReceiver handlerReceiver = new HandlerReceiverImpl();
+
+    @Override
+    public void addToStorage(String[] parsedData, Storage storage) {
+        for (String str : parsedData) {
+            String[] separatedData = str.split(COMA);
+            handlerReceiver
+                    .getHandler(separatedData[OPERATION])
+                    .updateDB(separatedData[FRUIT],
+                            Integer.parseInt(separatedData[QUANTITY]), storage);
+        }
+    }
```

Comment:
your service modifies data, after this - processes it, and brakes the Single responsibility principle. You need to use an additional model to work. 
![image](https://github.com/mate-academy/jv-fruit-shop/assets/32339836/54a93992-9b9b-4d6f-8c67-a1bfb9a7ad4a)